util/gvr.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+)
+
+func readYamlFile(filename string) (*unstructured.Unstructured, error) {
+	yamlFile, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read yaml file: %w", err)
+	}
+
+	var obj unstructured.Unstructured
+	if err := yaml.Unmarshal(yamlFile, &obj.Object); err != nil {
+		return nil, fmt.Errorf("failed to unmasrhsal into object: %w", err)
+	}
+
+	return &obj, nil
+}
+
+func CreateResource(filename string, dynamicClient *dynamic.DynamicClient, gvr schema.GroupVersionResource, namespace string, log *zap.Logger) error {
+	obj, err := readYamlFile(filename)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve configuration information: %w", err)
+	}
+
+	// Check if resource already exists
+	existing, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(context.TODO(), obj.GetName(), metav1.GetOptions{})
+	if err == nil {
+		log.Info("Resource already exists, skipping creation", zap.String("resource-name", existing.GetName()))
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to check if resource exists: %w", err)
+	}
+
+	result, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(context.TODO(), obj, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create resource: %w", err)
+	}
```

Comment:
Instead of using get to check existence and then create. Can we directly call Create and check if the error shows "already exists"?